Assets/Resources/Scripts/TransparentRay.cs

```
@@ -0,0 +1,121 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TransparentRay : MonoBehaviour {
+
+    /* To start the testing, turn the red player around and touch the blue thing in the starting room.  This will trigger the overhead camera with this script.
+     * The script currently works as desired, zooming in and out so everyone can be seen and centering on the middle of the players.
+     * The problem is that my code only works if the camera is directly overhead.
+     * The code should work for any angle and work as before except at an angle
+     * Thank you for your help and feel free to ask me any questions or to share work
+     * Sincerely, Kevin
+     */
+
+    private Stack<transparentObjectClass> transparentObjects=new Stack<transparentObjectClass>();
+    public Material tranparent;
+    public int resetCounter=0;
+    public GameObject[] players;
+
+    private int layerMask=1<<2;
+    // Use this for initialization
+    void Start () {
+        layerMask = ~(layerMask | (1 << 17));
+	}
+	
+	// This code only works if the camera is centered on the player
+    //Else add a seperate tag for the ground
+	void Update () {
+        RaycastHit hit;
+        //resets things to retry to make things better
+        if (resetCounter == 0)
+        {
+            //zooming- zooms in incase the players are close together
+            transform.Translate(Vector3.forward * 10);
+
+            try
+            {
+                while (transparentObjects.Peek() != null)
+                {
+                    transparentObjectClass reset = transparentObjects.Pop();
+                    reset.transparentObject.GetComponent<Renderer>().material = reset.originalMaterial;
+                    reset.transparentObject.layer = reset.originalLayer;
+                }
+            }
+            catch
```

empty catches are bad practice, some sort of error handling should be present